OpenRA.Mods.Common/Traits/Mobile.cs

```diff
@@ -743,11 +766,14 @@ public CPos NearestMoveableCell(CPos target, int minRange, int maxRange)
 			if (target.Layer != 0)
 				target = new CPos(target.X, target.Y);
 
-			if (CanEnterCell(target))
```

Comment:
Does our current code still contain the gotcha that `CanEnterCell(self.Location)` returns false, or did one of the previous movement PRs straighten that out?

If not, we'll need to check that all of the callers behave properly when `self.Location` is returned.

Reply:
Yes, this is still the case. The line below this was added to work around that.